site/compiler/index.js

```diff
@@ -1,14 +1,36 @@
-const pages = [
-  {
-    path: 'index.html',
-    body: 'Published at ' + (new Date().toString()),
-  },
-  {
-    path: '404.html',
-    body: 'Not found :(',
-  },
-];
-
-module.exports = function compileSite() {
-  return pages;
-};
+import React from 'react';
+import { renderToString } from 'react-dom/server';
+import { match, RouterContext } from 'react-router';
+import { uniq } from '../../lib/array';
```

Comment:
Again not sure what's the benefit of our own library for handling arrays. We'll likely eventually have switch to something like Ramda anyway, and then will have to remember to refactor existing code.